### PR TITLE
Improve shard-failed log messages.

### PR DESCRIPTION
**Pull request by DaveCTurner**
Previously:

    [WARN ][o.e.i.c.IndicesClusterStateService] [node_td2] [[test][0]] marking
        and sending shard failed due to [failed recovery]
    [WARN ][o.e.c.a.s.ShardStateAction] [node_tm0] [test][0] received shard
        failed for shard id [[test][0]], allocation id [m7t1rFIHTIKCsiecyLYvXA],
        primary term [0], message [failed recovery], failure
        [RecoveryFailedException[[test][0]: Recovery failed from
        {node_td1}{UKtBYHLJTwGWJ-bbSHxy_A}{MFiPmyjATnWl4oqp81GRqQ}{127.0.0.1}{127.0.0.1:9402}
        into
        {node_td2}{waqqel-jSSK2222vUOBVdg}{ZvUuFoC7S5eY2p6UqR45iA}{127.0.0.1}{127.0.0.1:9401}];
        nested:
        RemoteTransportException[[node_td1][127.0.0.1:53995][internal:index/shard/recovery/start_recovery]];
        nested: RecoveryEngineException[Phase[1] phase1 failed]; nested:
        RecoverFilesRecoveryException[Failed to transfer [15] files with total size of
        [10.9kb]]; nested:
        SendRequestTransportException[[node_td2][127.0.0.1:9401][internal:index/shard/recovery/file_chunk]];
        nested: NotSerializableExceptionWrapper[runtime_exception: Caused some
        truncated files for fun and profit]; ]
    [DEBUG][o.e.c.s.MasterService    ] [node_tm0] processing
        [shard-failed[shard id [[test][0]], allocation id [m7t1rFIHTIKCsiecyLYvXA],
        primary term [0], message [failed recovery], failure
        [RecoveryFailedException[[test][0]: Recovery failed from
        {node_td1}{UKtBYHLJTwGWJ-bbSHxy_A}{MFiPmyjATnWl4oqp81GRqQ}{127.0.0.1}{127.0.0.1:9402}
        into
        {node_td2}{waqqel-jSSK2222vUOBVdg}{ZvUuFoC7S5eY2p6UqR45iA}{127.0.0.1}{127.0.0.1:9401}];
        nested:
        RemoteTransportException[[node_td1][127.0.0.1:53995][internal:index/shard/recovery/start_recovery]];
        nested: RecoveryEngineException[Phase[1] phase1 failed]; nested:
        RecoverFilesRecoveryException[Failed to transfer [15] files with total size of
        [10.9kb]]; nested:
        SendRequestTransportException[[node_td2][127.0.0.1:9401][internal:index/shard/recovery/file_chunk]];
        nested: NotSerializableExceptionWrapper[runtime_exception: Caused some
        truncated files for fun and profit]; ]]]: execute

now

    [WARN ][o.e.i.c.IndicesClusterStateService] [node_td2] [[test][1]] marking
        and sending shard failed due to [failed recovery [runtime_exception: Caused
        some truncated files for fun and profit]]
    [WARN ][o.e.c.a.s.ShardStateAction] [node_tm0] [test][1] received shard
        failed for shard id [[test][1]] allocation id [OLqdfDFPTcC_74128virZw] primary
        term [0] message [failed recovery [runtime_exception: Caused some truncated
        files for fun and profit]]
    [DEBUG][o.e.c.s.MasterService    ] [node_tm0] processing [shard-failed
        shard id [[test][1]] allocation id [OLqdfDFPTcC_74128virZw] primary term [0]
        message [failed recovery [runtime_exception: Caused some truncated files for
        fun and profit]][shard id [[test][1]] allocation id [OLqdfDFPTcC_74128virZw]
        primary term [0] message [failed recovery [runtime_exception: Caused some
        truncated files for fun and profit]] failure
        [RecoveryFailedException[[test][1]: Recovery failed from
        {node_td1}{xg_rYbgNRaiiGi1Uy4-dlw}{sbcTdJfjSO2iJakJi4-5PQ}{127.0.0.1}{127.0.0.1:9401}
        into
        {node_td2}{RJab72gkTgCiuqYuTUDrvw}{DIyt0KSbSPeutYzWqPLyGw}{127.0.0.1}{127.0.0.1:9402}];
        nested:
        RemoteTransportException[[node_td1][127.0.0.1:53949][internal:index/shard/recovery/start_recovery]];
        nested: RecoveryEngineException[Phase[1] phase1 failed]; nested:
        RecoverFilesRecoveryException[Failed to transfer [15] files with total size of
        [5.5kb]]; nested:
        SendRequestTransportException[[node_td2][127.0.0.1:9402][internal:index/shard/recovery/file_chunk]];
        nested: NotSerializableExceptionWrapper[runtime_exception: Caused some
        truncated files for fun and profit]; ]]]: execute

### CODE REVIEW

**Review by jasontedor — CHANGES_REQUESTED** [review]
I don't think we need to log the whole request, only the shard ID?

**Comment by bleskes** [comment]
I agree that this is too verbose. On the other hand, going down to only the shard id is a shame too. I discussed this with @DaveCTurner and asked to proceed in two parts - the first is to add concise things here - like shard id, allocation id, primary term on the message (but not the full stack trace). The second is to improve the message for failed recoveries (right now it just says "recovery failed" but doesn't tell you what happened).

**Comment by DaveCTurner** [comment]
Toned the verbosity down a bit /cc @bleskes @jasontedor - is this more like what you were after?

File: core/src/main/java/org/elasticsearch/cluster/action/shard/ShardStateAction.java [hunk 1]
```diff
@@ -207,9 +207,11 @@ public void onTimeout(TimeValue timeout) {
 
         @Override
         public void messageReceived(ShardEntry request, TransportChannel channel) throws Exception {
-            logger.warn((Supplier<?>) () -> new ParameterizedMessage("{} received shard failed for {}", request.shardId, request), request.failure);
+            logger.warn((Supplier<?>) () -> new ParameterizedMessage("{} received shard failed for {}",
```
Comment (bleskes) [inline core/src/main/java/org/elasticsearch/cluster/action/shard/ShardStateAction.java]: This is just a log message. I think we want all the details.

File: core/src/main/java/org/elasticsearch/cluster/action/shard/ShardStateAction.java [hunk 2]
```diff
@@ -497,13 +499,23 @@ public void writeTo(StreamOutput out) throws IOException {
             out.writeException(failure);
         }
 
-        @Override
-        public String toString() {
-            List<String> components = new ArrayList<>(4);
+        private void addSummaryComponents(List<String> components) {
```
Comment (bleskes) [inline core/src/main/java/org/elasticsearch/cluster/action/shard/ShardStateAction.java]: NIT - I tend not to like that level of code reuse. It just makes it hard to see how the end result looks like. I also think we can just use string concat and be done with it.

**Review by bleskes — CHANGES_REQUESTED** [review]
I left one comment and one nit pick. Can you also share an example on how it the output looks now?

File: core/src/main/java/org/elasticsearch/cluster/action/shard/ShardStateAction.java (hunk 2 above)
Comment (DaveCTurner) [reply core/src/main/java/org/elasticsearch/cluster/action/shard/ShardStateAction.java]: Noted - no problem.

File: core/src/main/java/org/elasticsearch/cluster/action/shard/ShardStateAction.java (hunk 1 above)
Comment (DaveCTurner) [reply core/src/main/java/org/elasticsearch/cluster/action/shard/ShardStateAction.java]: The missing detail is just `request.failure` which seems to be logged anyway. This change just stops it logging that twice, I think.

Comment (bleskes) [reply core/src/main/java/org/elasticsearch/cluster/action/shard/ShardStateAction.java]: I missed the logging of the exception. All good then.

**Comment by DaveCTurner** [comment]
Ok, we've concluded that this isn't even solving the problem that we wanted to solve, so I'm closing this PR and will open another one.

**Pull request closed without merging**